netbox_cmdb/netbox_cmdb/models/interface.py

```diff
@@ -0,0 +1,123 @@
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from netbox.models import ChangeLoggedModel
+
+FEC_CHOICES = [
+    (None, "None"),
+    ("rs", "Reed Solomon"),
+    ("fc", "FireCode"),
+]
+
+LOGICAL_INTERFACE_TYPE_CHOICES = [
+    ("l1", "L1"),
+    ("l2", "L2"),
+    ("l3", "L3"),
+]
+LOGICAL_INTERFACE_MODE_CHOICES = [
+    (None, "None"),
+    ("access", "Access"),
+    ("tagged", "Tagged"),
+]
+
+
+class DeviceInterface(ChangeLoggedModel):
+    """A device interface configuration."""
+
+    name = models.CharField(max_length=100)
+    device = models.ForeignKey(
+        to="dcim.Device",
+        on_delete=models.CASCADE,
+        related_name="%(class)sdevice",
+        null=False,
+        blank=False,
+    )
+    autonegotiation = models.BooleanField(default=True)
+    speed = models.PositiveIntegerField(
+        validators=[MinValueValidator(1), MaxValueValidator(800000000)],
+        blank=True,
+        null=True,
+        help_text="Interface speed in kB/s",
+    )
+    fec = models.CharField(
+        choices=FEC_CHOICES,
+        max_length=5,
+        blank=True,
+        null=True,
+    )
+    description = models.CharField(max_length=100, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.device.name}--{self.name}"
+
+    class Meta:
+        unique_together = ("device", "name")
+
+
+class LogicalInterface(ChangeLoggedModel):
+    """A logical interface configuration."""
+
+    index = models.PositiveSmallIntegerField()
+    parent_interface = models.ForeignKey(
+        to="DeviceInterface",
+        related_name="%(class)s",
+        on_delete=models.CASCADE,
+        blank=True,
```

Comment:
In which cases do we allow to have an empty parent_interface? Loopback interfaces ?

Reply:
as there is no name field for the logical interface, it seems to be mandatory to have a parent_interface